### DISCUSSION TITLE
How to get gun fire data?

### DISCUSSION THREAD

**Discussion by DiegoZoracKy** [saul/demofile#247] [Questions] [answered]
Hi @saul,

First, thanks at  for this awesome piece of work. Is it possible to get data from gun fires, bullets trajectory or anything like that? I couldn't find info about it on the readme, not even searching through the issues.

**Comment by Technoblazed** [comment]
If you add a hook for the `player_shoot` event, the object for the event includes a player location with vel/pos so you can figure out where the user is shooting.

**Comment by saul** [comment]
@DiegoZoracKy has @Technoblazed's answer helped you, or do you have more questions?

**Comment by DiegoZoracKy** [comment]
Thanks @Technoblazed and @saul. I'll try to take a look at this event. Is there some place where I can find a list of all events being emitted?

**Comment by DiegoZoracKy** [comment]
Nevermind. I believe I found it:

https://github.com/saul/demofile/blob/6762aeb646383abbc0e5cbca26a23174e8753fcd/src/eventtypes.ts

**Comment by DiegoZoracKy** [comment]
Where should I listen for this event? I tried at gameEvent and entities but I got nothing from them.

**Comment by saul** [comment]
@diegozoracky what code are you using that isn’t working?

**Comment by Technoblazed** [comment]
https://wiki.alliedmods.net/Game_Events_(Source)
https://wiki.alliedmods.net/Counter-Strike:_Global_Offensive_Events

**Comment by DiegoZoracKy** [comment]
> @DiegoZoracKy what code are you using that isn’t working?

@saul, I tried with:

`demoFile.gameEvents.on('player_shoot', ...` and `demoFile.entities.on('player_shoot', ...`. 

But based on the list @Technoblazed posted above, it seems that the right event is `weapon_fired`.

**Comment by DiegoZoracKy** [comment]
I've found the event **bullet_impact** on https://wiki.alliedmods.net/Counter-Strike:_Global_Offensive_Events , but I can't get any data when trying to listen at `demoFile.gameEvents.on('bullet_impact')`. Am I listening to it in a wrong way or is this event not being emmited for some known reason?

**Comment by saul** [comment, marked as answer]
Lots of events are not written to demo files, and so won't be available. It's hit and miss knowing which ones are recorded or not. 

I'm pretty sure `weapon_fire` is networked, though. Please comment if you can't get what you want from that event.
